### PR TITLE
Move config file

### PR DESCRIPTION
**Pull request by msaggiorato**
## Summary:

Partially address https://github.com/Varying-Vagrant-Vagrants/VVV/issues/1922

Moves config file to config folder, and perform the migration of the old version to the new one. Also changed comments and puts to reflect the new location.

## Checks

 - [x] I've tested this PR with Vagrant **vXX** and VirtualBox **vXX** on **Operating System**
 - [x] This PR is for the `develop` branch not the `master` branch
 - [ ] I've updated the changelog
 - [x] This PR is complete and ready for review

### CODE REVIEW

**Comment by msaggiorato** [comment]
Not sure if this labeler issue is a problem on the PR, please let me know. I've provisioned with these changes and everything worked (and vvv-custom was moved accordingly).

File: Vagrantfile [hunk 1]
```diff
@@ -553,16 +557,13 @@ sudo chown -R vagrant:vagrant /vagrant
...
-rm -f /vagrant/vvv-custom.yml
 
 touch /vagrant/provisioned_at
 echo `date "+%Y%m%d-%H%M%S"` > /vagrant/provisioned_at
 
 # copy over version and config files
 cp -f /home/vagrant/version /vagrant
-cp -f /home/vagrant/vvv-custom.yml /vagrant
 
-sudo chmod 0644 /vagrant/vvv-custom.yml
```
Comment (Mte90) [inline Vagrantfile]: Wondering if we put that to avoid some conflicts or other issues on provisions on some systems. what do you think @tomjn?

**Comment by tomjn** [comment]
That fixed an issue Paul Schreiber was having with the file being unreadable

**Comment by Mte90** [comment]
So is better to leave that chmod to avoid any issue

File: Vagrantfile (hunk 1 above)
Comment (Mte90) [reply Vagrantfile]: @msaggiorato better to give back that command :-) and is ready to be tested

**Comment by msaggiorato** [comment]
But now that the file is on the config folder, i'd say it should be readable with no problem. If the file is not readable, then likely the whole provisioner wont work as the config folder is quite critical to it.

Am I wrong?

Before this change, the vvv-custom and vvv-config files were provisioned as files, not synced folders.

Correct me if I'm wrong

**Comment by tomjn** [comment]
> Before this change, the vvv-custom and vvv-config files were provisioned as files, not synced folders.

Those files still have to be copied into place in `/vagrant`, as a record of what was provisioned. That way even if `config/config.yml` changes we know that if they're different a reprovision is needed.

As for why that file was unreadable and needed the `chmod`, ¯\_(ツ)_/¯, I never encountered the issue myself, but someone did, and for them it was necessary. Without it, the dashboard crashed.

File: .gitignore [hunk 2]
```diff
@@ -11,7 +11,7 @@ Icon\r\r
 
 # This is a file intended for hooking in a custom Vagrant configuration on up
 /Customfile
-/vvv-custom.yml
```
Comment (tomjn) [inline .gitignore]: I think we should still ignore this file

File: Vagrantfile [hunk 3]
```diff
@@ -125,9 +125,16 @@ end
 
 # Load the config file before the second section of the splash screen
 
-if File.file?(File.join(vagrant_dir, 'vvv-custom.yml')) == false then
-  puts "#{yellow}Copying #{red}vvv-config.yml#{yellow} to #{green}vvv-custom.yml#{yellow}\nIMPORTANT NOTE: Make all modifications to #{green}vvv-custom.yml#{yellow} in future so that they are not lost when VVV updates.#{creset}\n\n"
-  FileUtils.cp( File.join(vagrant_dir, 'vvv-config.yml'), File.join(vagrant_dir, 'vvv-custom.yml') )
+vvv_config_file = File.join(vagrant_dir, 'config/config.yml')
```
Comment (tomjn) [inline Vagrantfile]: We also need to copy `vvv-custom.yml` to `config/config.yml` if it exists for backwards compatibility

Comment (Mte90) [reply Vagrantfile]: uh I missed that

**Comment by tomjn** [comment]
We also need to update the changelog, aside from the bullet point, there'll need to be a small paragraph explaining the change

File: Vagrantfile (hunk 3 above)
Comment (msaggiorato) [reply Vagrantfile]: But it is copying `vvv-custom.yml` to `config/config.yml` for backwards compatibilty.

What am I missing?

**Comment by msaggiorato** [comment]
> > Before this change, the vvv-custom and vvv-config files were provisioned as files, not synced folders.
> 
> Those files still have to be copied into place in `/vagrant`, as a record of what was provisioned. That way even if `config/config.yml` changes we know that if they're different a reprovision is needed.
> 
> As for why that file was unreadable and needed the `chmod`, ¯_(ツ)_/¯, I never encountered the issue myself, but someone did, and for them it was necessary. Without it, the dashboard crashed.

If we need to do this, should we always be reading `/vagrant/config.yml` then?

File: Vagrantfile (hunk 3 above)
Comment (tomjn) [reply Vagrantfile]: hmm I've misunderstood, it needs to copy `config/default-config.yml` to `config/config.yml` too


```
if config/config.yml doesn't exist
    if vvv-custom.yml exists
        copy that to config/config.yml
    else
        copy config/default-config.yml to config/config.yml
```

It should also mention the filenames explicitly so we don't confuse any users

Comment (msaggiorato) [reply Vagrantfile]: This is what's it's currently doing with my changes

```
if vvv-custom.yml exists
   copy that to config/config.yml

if config/config.yml doesn't exist
    copy config/default-config.yml to config/config.yml
```

Pretty much the same.

Comment (tomjn) [reply Vagrantfile]: I've committed an update that refactors that bit

Comment (msaggiorato) [reply Vagrantfile]: Awesome, thanks!

**Comment by tomjn** [comment]
> If we need to do this, should we always be reading /vagrant/config.yml then?

That depends on the use case, but for the provisioners purposes it should always be the original. Eventually it'll be Ansible reading it in anyway.

I think all we need to do now is update the changelog. I think we should hold off a bit on merging it in though so we don't catch users on the `develop` branch offguard, ideally we do this shortly before a 3.2 release so we can update the documentation at the same time

**Comment by msaggiorato** [comment]
Awesome, should I update the changelog? Or you have it?

**Comment by tomjn** [comment]
if you can wrangle the changelog? :)

**Comment by msaggiorato** [comment]
Should we do the changelog update separately? Seems like there's a bunch of things that changed also outside of this PR since it was updated.

**Comment by msaggiorato** [comment]
> Also, now that the original config file is in a synced folder again, we can remove the vagrant provisioner that copies those files to `/vagrant` and just do it bash, see #1958

Pushed a change that address #1958. Could there be any problems with provision-pre.sh? What do people usually do in there?

**Comment by tomjn** [comment]
Not many people use `provision-pre.sh`, considering at some point there won't be a `provision.sh` anyway I'm fine with that.

Also we should update the changelog in this PR, a rebase on to `develop` to bring in the newer changes should resolve the conflict, afterall that file is only being moved/renamed

**Comment by msaggiorato** [comment]
Rebased

**Comment by tomjn** [comment]
We're a WIP docs update PR, we'll need one for the dashboard too

**Pull request merged**